src/image.rs

```diff
@@ -56,12 +58,21 @@ impl fmt::Display for ImageError {
     }
 }
 
-impl FromError<old_io::IoError> for ImageError {
-    fn from_error(err: old_io::IoError) -> ImageError {
+impl FromError<io::Error> for ImageError {
+    fn from_error(err: io::Error) -> ImageError {
         ImageError::IoError(err)
     }
 }
 
+impl FromError<byteorder::Error> for ImageError {
+    fn from_error(err: byteorder::Error) -> ImageError {
+        match err {
+            byteorder::Error::UnexpectedEOF => ImageError::ImageEnd,
```

Comment:
The underlying problem is that `ErrorKind::EndOfFile` no longer exists in new I/O.
If you want to read at least N bytes, but EOF happens before, it used to return this error.

That's why we need a replacement for it (here and at some other places). I picked `ImageEnd` but was not really sure about it.